src/lib/operators/table_scan_implementation.hpp

```diff
@@ -0,0 +1,362 @@
+            }
+          }
+
+          result_segment =
+              std::make_shared<ReferenceSegment>(reference_source_segment->referenced_table(),
+                                                 reference_source_segment->referenced_column_id(), pos_list_ptr);
+        }
+      } else {
+        if (pos_list_for_value_and_dictionary_segment == nullptr) {
+          pos_list_for_value_and_dictionary_segment = std::make_shared<PosList>();
+
+          if (_add_all_chunk_offsets) {
+            ChunkOffset source_chunk_size = source_chunk.size();
+            pos_list_for_value_and_dictionary_segment->reserve(source_chunk_size);
+            for (ChunkOffset chunk_offset = 0; chunk_offset < source_chunk_size; ++chunk_offset) {
+              pos_list_for_value_and_dictionary_segment->emplace_back(chunk_id, chunk_offset);
+            }
+          } else {
+            pos_list_for_value_and_dictionary_segment->reserve(_chunk_offsets_to_add_to_result_table.size());
+
+            for (const ChunkOffset chunk_offset : _chunk_offsets_to_add_to_result_table) {
+              pos_list_for_value_and_dictionary_segment->emplace_back(chunk_id, chunk_offset);
+            }
+          }
+        }
+
+        result_segment =
+            std::make_shared<ReferenceSegment>(_table, column_id, pos_list_for_value_and_dictionary_segment);
+      }
+
+      result_chunk.add_segment(result_segment);
+    }
+
+    _chunk_offsets_to_add_to_result_table.clear();
+    _result_table->emplace_chunk(std::move(result_chunk));
+    _add_all_chunk_offsets = false;
+  }
+
+  std::function<bool(T)> _get_scan_type_comparator() {
+    switch (_scan_type) {
+      case ScanType::OpEquals:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 == _typed_search_value; };
+      case ScanType::OpNotEquals:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 != _typed_search_value; };
+
+      case ScanType::OpLessThan:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 < _typed_search_value; };
+      case ScanType::OpLessThanEquals:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 <= _typed_search_value; };
+
+      case ScanType::OpGreaterThan:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 > _typed_search_value; };
+      case ScanType::OpGreaterThanEquals:
+        return [&_typed_search_value = _typed_search_value](const T& val1) { return val1 >= _typed_search_value; };
+    }
+
+    throw std::logic_error("Unhandled scan type in _get_scan_type_comparator");
+  }
+
+  struct DictionarySegmentProcessingFlags {
+    DictionarySegmentProcessingFlags(const std::shared_ptr<DictionarySegment<T>> segment, const ScanType& _scan_type,
+                                     const T& _typed_search_value) {
+      ValueID matching_value_id;
```

Comment:
This does not seem to be used outside of the switch statement. Try to keep the scopes small.